core/src/main/java/org/apache/james/core/quota/QuotaUsageValue.java

```diff
@@ -0,0 +1,44 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.core.quota;
+
+import java.util.Optional;
+
+public interface QuotaUsageValue<T extends QuotaUsageValue<T, U>, U extends QuotaLimitValue<U>> {
+
+    static boolean isValidValue(Optional<Long> value) {
+        return !value.isPresent() || value.get() >= -1;
+    }
+
+    long asLong();
+
+    boolean isLimited();
+
+    default boolean isUnlimited() {
```

Comment:
Unlimited have no meaning for "usage", this do not make sens!

Server resources usage are necessarily limited even though  limit is unilited.

Reply:
Ok fixed on next commit